cnf-certification-test/chaostesting/suite.go

```diff
@@ -0,0 +1,280 @@
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	// timeout for eventually call
+	RequestTimeout     = 120 * time.Second
+	Deployment         = "deployment"
+	ServiceAccountFile = "chaostesting/servicAccount.yaml"
+	ExperimentFile     = "chaostesting/deleteExperment.yaml"
+	chaosEngineFile    = "chaostesting/chaosEngine.yaml"
+	chaosname          = "pod-delete"
+	completedResult    = "completed"
+)
+
+var _ = ginkgo.Describe(common.ChaosTesting, func() {
+	var env provider.TestEnvironment
+
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+
+	// container security context: privileged escalation
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodDeleteIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testPodDelete(&env)
+	})
+
+})
+
+func testPodDelete(env *provider.TestEnvironment) {
+	for _, dep := range env.Deployments {
+		namespace := dep.ObjectMeta.Namespace
+		label := "app=" + dep.Spec.Template.ObjectMeta.Labels["app"]
+		fileName, err := applyTemplate(label, Deployment, namespace, ExperimentFile)
+		if err != nil {
+			logrus.Debugf("cant create the file of the test: %e", err)
+			ginkgo.Fail(fmt.Sprintf("cant create the file of the test: %e.", err))
+		}
+		if _, err = createResource(fileName); err != nil {
+			ginkgo.Fail(fmt.Sprintf("%e error create the chaos experment resources.", err))
+		}
+		fileName, err = applyTemplate(label, Deployment, namespace, ServiceAccountFile)
+		if err != nil {
+			logrus.Debugf("cant create the file of the test: %e", err)
+			ginkgo.Fail(fmt.Sprintf("cant create the file of the test: %e.", err))
+		}
+		if _, err = createResource(fileName); err != nil {
+			ginkgo.Fail(fmt.Sprintf("error create the service account: %e .", err))
+		}
+		fileName, err = applyTemplate(label, Deployment, namespace, chaosEngineFile)
+		if err != nil {
+			logrus.Debugf("cant create the file of the test: %e", err)
+			ginkgo.Fail(fmt.Sprintf("cant create the file of the test: %e.", err))
+		}
+		// create the chaos engine for every deployment in the cluster
+		if _, err = createResource(fileName); err != nil {
+			ginkgo.Fail(fmt.Sprintf("%e error create the chaos engine.", err))
+		}
+		time.Sleep(1 * time.Second)
+		completed := waitForTestFinish(RequestTimeout)
+		if completed {
+			if finalResult, result := returnResult(); finalResult {
+				// delete the chaos engin crd
+				deleteAllResources(namespace)
+			} else {
+				logrus.Debugf("test completed but it failed with reason %s", result)
+				ginkgo.Fail(fmt.Sprintf("test completed but it failed with reason %s", result))
+			}
+		} else {
+			logrus.Debug("test failed to be completed")
+			ginkgo.Fail("test failed to be completed")
```

Comment:
As the ginkgo.Failf will abort the current execution of the TC, you can omit the if/else if you just check like this:
```
    if !completed {
	    logrus.Debug("test failed to be completed")
	    ginkgo.Fail("test failed to be completed")
	}
	if finalResult, result := returnResult(); finalResult {
	    ...
    }
```